src/main/java/com/google/sps/servlets/IndexServlet.java

```diff
@@ -0,0 +1,147 @@
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+    TemplateData templateData = getTemplateData(pool);
+
+    // Add variables and render the page
+    req.setAttribute("tabCount", templateData.tabCount);
+    req.setAttribute("spaceCount", templateData.spaceCount);
+    req.setAttribute("recentVotes", templateData.recentVotes);
+    req.getRequestDispatcher("/index.jsp").forward(req, resp);
+  }
+
+  // Used to validate user input. All user provided data should be validated and sanitized before
+  // being used something like a SQL query. Returns null if invalid.
+  @Nullable
+  private String validateTeam(String input) {
+    if (input != null) {
+      input = input.toUpperCase(Locale.ENGLISH);
+      // Must be either "TABS" or "SPACES"
+      if (!"TABS".equals(input) && !"SPACES".equals(input)) {
+        return null;
+      }
+    }
+    return input;
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    // Get the team from the request and record the time of the vote.
+    String team = validateTeam(req.getParameter("team"));
+    Timestamp now = new Timestamp(new Date().getTime());
+    if (team == null) {
+      resp.setStatus(400);
+      resp.getWriter().append("Invalid team specified.");
+      return;
+    }
+
+    // Reuse the pool that was created in the ContextListener when the Servlet started.
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+    // [START cloud_sql_mysql_servlet_connection]
+    // Using a try-with-resources statement ensures that the connection is always released back
+    // into the pool at the end of the statement (even if an error occurs)
+    try (Connection conn = pool.getConnection()) {
+
+      // PreparedStatements can be more efficient and project against injections.
+      String stmt = "INSERT INTO votes (time_cast, candidate) VALUES (?, ?);";
+      try (PreparedStatement voteStmt = conn.prepareStatement(stmt);) {
+        voteStmt.setTimestamp(1, now);
+        voteStmt.setString(2, team);
+
+        // Finally, execute the statement. If it fails, an error will be thrown.
+        voteStmt.execute();
+      }
+    } catch (SQLException ex) {
+      // If something goes wrong, handle the error in this section. This might involve retrying or
+      // adjusting parameters depending on the situation.
+      // [START_EXCLUDE]
+      LOGGER.log(Level.WARNING, "Error while attempting to submit vote.", ex);
+      resp.setStatus(500);
+      resp.getWriter()
+          .write(
+              "Unable to successfully cast vote! Please check the application "
+                  + "logs for more details.");
+      // [END_EXCLUDE]
+    }
+    // [END cloud_sql_mysql_servlet_connection]
+
+    resp.setStatus(200);
+    resp.getWriter().printf("Vote successfully cast for '%s' at time %s!%n", team, now);
+  }*/
+}
```

Comment:
Are these comments supposed to be here? It would be cleaner to remove comments

Reply:
test purposes.

Reply:
I deleted it in the last version btw haha